app/experimenter/nimbus-ui/src/components/FormBranches/index.tsx

```diff
@@ -20,27 +21,53 @@ import {
   FormBranchesSaveState,
   REFERENCE_BRANCH_IDX,
   AnnotatedBranch,
-  extractSaveState,
 } from "./reducer";
 
 export const FormBranches = ({
+  isLoading,
   experiment,
   featureConfig,
   onSave,
   onNext,
 }: {
+  isLoading: boolean;
   experiment: getExperiment_experimentBySlug;
   featureConfig: getConfig_nimbusConfig["featureConfig"];
-  onSave: (state: FormBranchesSaveState) => void;
+  onSave: (
+    state: FormBranchesSaveState,
+    setSubmitErrors: Function,
+    clearSubmitErrors: Function,
+  ) => void;
   onNext: () => void;
 }) => {
-  const [formBranchesState, dispatch] = useFormBranchesReducer(experiment);
-  const {
-    featureConfig: experimentFeatureConfig,
-    referenceBranch,
-    treatmentBranches,
-    equalRatio,
-  } = formBranchesState;
+  const [
+    {
+      featureConfig: experimentFeatureConfig,
+      referenceBranch,
+      treatmentBranches,
+      equalRatio,
+      globalErrors,
+    },
+    extractSaveState,
+    dispatch,
+  ] = useFormBranchesReducer(experiment);
+
+  const isSaveDisabled =
+    isLoading ||
+    !referenceBranch ||
+    !referenceBranch.isValid ||
+    !treatmentBranches ||
+    !treatmentBranches.every((branch) => branch.isValid);
```

Comment:
You'll want to pull in the latest from main, it looks like we've stopped disabling the Save button entirely except in loading states.

Reply:
I'll be tweaking this conditional, but we can't save arbitrary state without constraint. If the branch data is not valid at least with respect to the GQL types, we'll get a 400 Bad Request on submission of the mutation.